feat: Implement cloud storage provider backups system with OAuth support

vishalkadam47 opened the pull request:
Add support for multiple cloud storage providers (Google Drive, Dropbox, Box, FTP, SFTP) for database and web server backups. This implementation follows the same pattern as S3 backups while handling provider-specific authentication and path construction.
- OAuth-based authentication for cloud providers (Drive, Dropbox, Box)
- Credential-based authentication for FTP/SFTP
- add `withSilentTokenRefresh` automatically refreshes expired OAuth tokens and retries cloud storage operations.
If token is invalid/expired then it initiates new OAuth flow, I have also added a refresh token button which does the same when triggered from `dashboard/settings/server`

![Screenshot at 2025-05-31 18-13-54](https://github.com/user-attachments/assets/79fbd230-151b-4a5e-8309-511d391b486e)
![Screenshot at 2025-05-31 18-20-52](https://github.com/user-attachments/assets/f77f52dd-330c-4bcc-a932-36c5e197d1e9)
![Screenshot at 2025-05-31 18-58-01](https://github.com/user-attachments/assets/2a34796e-1158-4844-b34e-499347fa16ee)
![Screenshot at 2025-05-31 19-02-38](https://github.com/user-attachments/assets/1b7b3773-4e87-4aa0-90ec-2ba929e9e8e2)

vishalkadam47 committed:
feat: Implement cloud storage provider backups system with OAuth support

vishalkadam47 committed:
refactor: streamline cloud storage backup and restore processes by simplifying path construction and enhancing logging

vishalkadam47 committed:
refactor: clean up cloud storage components

vishalkadam47 committed:
Merge branch 'Dokploy:canary' into feat/cloud-storage-providers

vishalkadam47 committed:
Merge branch 'Dokploy:canary' into feat/cloud-storage-providers

Siumauricio committed:
Remove cloud storage tables and associated metadata for "pale_guardsmen" from the database schema.

Siumauricio committed:
Merge branch 'canary' into feat/cloud-storage-providers

Siumauricio committed:
feat(cloud-storage): add cloud storage destination and backup tables

- Created new SQL tables for managing cloud storage destinations and backups, including necessary fields and constraints.
- Updated the journal and snapshot metadata to reflect the addition of these new tables for version 0098.

Siumauricio reviewed (commented):
- Would be ideal to leave the primary button enable, and when the user tries to submit it automatically will show what fields are missing, I didn't know what was required and what no
![Screenshot 2025-06-22 at 12 03 14 AM](https://github.com/user-attachments/assets/a1013432-4be4-4d44-b2e3-daa1c404c3b0)

- I also think that changing between providers example: let's suppose you have a filled form from FTP provider and then you change to SFTP you still have the filled values, so I believe it should clear when you change between different providers 

- On the WebServers backups, can you check the UI? 
![Screenshot 2025-06-22 at 12 05 33 AM](https://github.com/user-attachments/assets/a42e1b6c-fa14-489b-9cf1-6bf165f84bf1)

- Also I think we should have the same UI when creating, updating or deleting the backups, because currently we have two differents UIs for the same form, so maybe we can reuse the form, and also when we show the backups it will be ideal to show everything in the same list not in separated

- Also when running a backup I have this error, I don't see the implementation of the backups with this new destinations
![Screenshot 2025-06-22 at 12 11 09 AM](https://github.com/user-attachments/assets/0a13eb58-e761-4c90-88ab-b20f2b972d39)

vishalkadam47 committed:
Drop migration 0098_majestic_white_queen to avoid conflicts

- Removed 0098_majestic_white_queen.sql
- Removed 0098_snapshot.json
- Updated _journal.json
- Will recreate as 0104 after merge

vishalkadam47 committed:
Merge canary into feat/cloud-storage-providers

Successfully merged canary branch with cloud storage providers feature:
- Resolved conflict in handle-destinations.tsx
- Added new migrations 0098-0103 from canary
- Integrated volume backup features
- Updated dependencies and configurations
- No migration conflicts after dropping 0098_majestic_white_queen

vishalkadam47 committed:
refactor: unify backup form UI

vishalkadam47 committed:
Merge branch 'Dokploy:canary' into feat/cloud-storage-providers

vishalkadam47 committed:
Remove entry for "0104_flowery_felicia_hardy" from the journal in meta data

vishalkadam47 committed:
Merge remote-tracking branch 'upstream/canary' into feat/cloud-storage-providers

vishalkadam47 committed:
Merge branch 'upstream/canary' into feat/cloud-storage-providers

vishalkadam47 committed:
Merge branch 'Dokploy:canary' into feat/cloud-storage-providers

dennisimoo commented:
Anyone know when this is gonna get merged? Really looking forward to this.

besteFinanzen commented:
Hi, can we get this merged?

dennisimoo commented:
This is just never gonna get merged

besteFinanzen commented:
> merge ts ong

?

IksGarG commented:
Would love to see this in prod 🙏
